core-bundle/src/Controller/ContentElement/AbstractContentElementController.php

```diff
@@ -60,5 +71,36 @@ protected function addSharedMaxAgeToResponse(Response $response, ContentModel $m
         $response->setSharedMaxAge(min($min));
     }
 
+    protected function getUpcomingResponse(Request $request, ContentModel $model): ?Response
+    {
+        if (!$this->container->get('contao.routing.scope_matcher')->isFrontendRequest($request)) {
+            return null;
+        }
+
+        if ($this->container->get('contao.security.token_checker')->isPreviewMode()) {
+            return null;
+        }
+
+        $time = time();
+
+        if ('' !== $model->stop && $model->stop < $time) {
```

Comment:
```suggestion
        if ($model->invisible || ('' !== $model->stop && $model->stop < $time)) {
```

Checking `invisible` flag here would make https://github.com/contao/contao/pull/2867 superfluous.